numpy/ma/tests/test_core.py

```diff
@@ -1327,14 +1327,32 @@ def test_eq_on_structured(self):
         test = (a == a)
         assert_equal(test, [True, True])
         assert_equal(test.mask, [False, False])
+        test = (a == a[0])
+        assert_equal(test, [True, False])
```

Comment:
You're presuming that numpy functions correctly account for subclasses.... ;-)
```
In [2]: test = np.ma.MaskedArray([True, True], mask=[False, True])
In [3]: assert_equal(test, [True, True])
In [4]: test = np.ma.MaskedArray([True, False], mask=[False, True])
In [5]: assert_equal(test, [True, False])
```
(they should, of course, but that's for another time....)